Revert "Revert "Adds default sql indexes and table optimisations""

Jordie0608 opened the pull request:
See #24277, reverted pending MSO's avaliability so we don't break the servers. Don't merge this time until then.

Commit:
boxes and icon overlays

PlayerDeer committed:
Updates 40mm, 9mm, 10mm, and 45mm sprites

PlayerDeer committed:
small tweak

Commit:
neew booxxeesss

Commit:
can you explain in drug boats please

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into more_3/4

# Conflicts:
#	icons/obj/storage.dmi

Commit:
pls

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into more_3/4

Commit:
Merge branch 'more_3/4' of https://github.com/Militaires/tgstation into more_3/4

# Conflicts:
#	icons/obj/storage.dmi

Commit:
why god

Commit:
Disables sound variance on cryo tube alert sound

Cyberboss committed:
Sorted client time timers

Cyberboss committed:
Improves the report issue button

Commit:
fixy

ChangelingRain committed:
You can now have multiple research scanners active at a time

Jordie0608 committed:
Revert "Revert "Adds default sql indexes and table optimisations""

Commit:
Fixes missing R&D shutter near public autolathe (#24312)

tgstation-server committed:
Automatic changelog generation for PR #24312 [ci skip]

Commit:
weeeeooooweeoowowowowow i deed it yah wah one up!!!

Commit:
minor edit

Cyberboss committed:
Overlay queuing (#23922)

* Overlay queuing

* Fix SS flags

* Don't copy on assignment

* Flags processing

* Fix icon_smoothing

* MSO's helper proc

* Legacy detection

* Make it work

* Fixes shitcode

* Fix the flag

* |= -> +=

* OK, how did I fuck that up?

* shitcode

* Conditional assoc queue while initializing

* Cleanup everything

* Orange meme

* This isn't perfect, but its the best byond will give us.

* forgot about dir

* oh ya

* This was litterally the last thing i did last night before heading to bed

You can tell can't you?

* Fixes various shit

* Let's not ever pause

* Fix the flag

* Cleaned up some missing shit. Added image dummys

* Remove the one usage of FPRINT

* Jesus get rid of this

MrPerson committed:
Ion storm improvements (#24223)

* Ion storm improvements

Ion storms have several new additions:
25% chance to flatly replace the AI's core lawset with something random in the config. Suddenly the AI is Corporate, deal w/ it.
10% chance to delete one of the AI's core or supplied laws. Hope you treated the AI well without its precious law 1 to protect your sorry ass.
10% chance that, instead of adding a random law, it will instead replace one of the AI's existing core or supplied laws with the ion law. Otherwise, it adds the generated law as normal. There's still a 100% chance of getting a generated ion law.

All of these stack so you could wind up going from Asimov to Paladin w/o the first law and w/ the last law replaced with THE SHUTTLE CANNOT BE CALLED DUE TO FIVE NINJAS. All the values are easy to tweak if you guys want them higher or lower or whatever.

Custom admin-sent and other fake ion storms (devils) will just add the law and have no chance of doing any of the bonus stuff.

Removed the admin verb to send an ion storm since you can just use the events panel.

Cleaned up some of the law-adding backend. Hopefully there's no double showing of the AI's laws after a lawchange as a result of this.

* Everyday I'm shufflin'

tgstation-server committed:
Automatic changelog generation for PR #24223 [ci skip]

AnonymousNow committed:
Adds Medical HUDGlasses (#24096)

* Adds Medical HUDGlasses

Adds sunglasses with a built-in medical HUD. Functions as you would expect them to, tested locally with no issues, comes in a stylish blue.

I've wanted these for literally years, but I've only sat down and worked out vaguely how items and PRs work this week, so here they are. They haven't been added to any map yet - that's a debate itself - but I'm hoping to find an opportunity to test them, perhaps being ERT Medical Officer standard equipment before a single pair is put in medbay to fight over.

Lemme know what you think.

* Hopefully moves hud.dm to its intended place.

Root folder indeed. WTH, Git?

* Glasses.dmi in the wrong place

Yep.

* eyes.dmi in the wrong place

Alright.

* Re-adds eyes.dmi in the right place

Well, guess Git's not as smart as I hoped.

* Re-adds glasses.dmi in the right place

LET US NEVER SPEAK OF THIS AGAIN.

* Deletes the old hud.dm to take place for the new

This'll all be cleaner next time.

* Moves hud.dm to where it should have been in the first place

Second PR ever, everybody!

* Adds Inhands for sunhudmed and sunhudsec

Oops. Forgot the inhand sprites. While I'm here, let's give security HUDshades their own inhands - they use the generic shade inhands otherwise, it seems.

* Tweaks shading on sunhudmed

Makes it like my original concept image which I just found again (over four years old!); now it makes a little more sense and looks better.

* Tweaks colour on sunhudmed to be a little more distinct

tgstation-server committed:
Automatic changelog generation for PR #24096 [ci skip]

Cyberboss committed:
Cleans up new lobby timing (#24250)

* Cleans up new lobby timing

* Changes

Cyberboss committed:
Makes game start smoother (#23552)

* Makes game start smoother

* Hide the sausage making

* Fix the shit

* Fixes AI's losing their name prefs

tgstation-server committed:
Automatic changelog generation for PR #23552 [ci skip]

ChangelingRain committed:
Improves catwalk sprites (#24333)

* Improves catwalk sprites

* Subdued alt

Cyberboss committed:
Does a thing for @RemieRichards

Commit:
why do we even do it like this, why would we not make them restock the maximum amount possible just by default

Commit:
tyrexvj ifkrkzfe

Cyberboss committed:
NobodyReviewsNobodyReviewsNobodyReviews (#24360)

NobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviewsNobodyReviews

TalkingCactus committed:
fixes backpacks being = 1 in character creation (#24314)

tgstation-server committed:
Automatic changelog generation for PR #24314 [ci skip]

silicons committed:
The floor really can't seem to mount the NekoOS... (#24315)

* Update robot.dm

* Update human.dm

Commit:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Fixes the fire overlay never being cut (#24354)

tgstation-server committed:
Automatic changelog generation for PR #24354 [ci skip]

Cyberboss committed:
Overlay/Roundstart fixpack (#24356)

* Fixes roundstart graphical lag

* Fixes manifests

* >Expecting searching lists to be fast

* Here's where they needed flushing

* THE DATACORE IS ANNOYING REEE

* <Commit name>

tgstation-server committed:
Automatic changelog generation for PR #24356 [ci skip]

nfreader commented:
This'll smash part of my stats app, but should be easy enough to catch up on the changes. Memo me on IRC if it looks like I'm asleep at the wheel.

ChangelingRain committed:
Cult messages will now always include the user's real name

Commit:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Revert "Overlay/Roundstart fixpack (#24356)"

This reverts commit bf42d769e2407491116b7f02d9e366cd97e69caa.

Cyberboss committed:
Revert "Makes game start smoother (#23552)"

This reverts commit 3ab308aa0f8f3be657fce5fe1eb0dbb588292aa4.

imtakingabreakdontatme committed:
Fixes invisible augmetic limbs

Cheridan committed:
Merge pull request #24369 from KorPhaeron/arms2

[s]Fixes augments being invisible

Cheridan committed:
Merge pull request #24334 from Cyberboss/BetterRporting

Improves the report issue button

coiax committed:
Taste (#24323)

* Adding taste to reagents

* Toxins mostly taste bitter.

* The taste of medicine is mostly bitterness

* Drugs taste bitter too

* It turns out pyrotechnics mostly taste like burning

* Alcohol and blob tastes

* Drinks taste mostly of drinks

* And the food reagents are the last, but not least

* Actual taste code

* You can now taste freshly spawned carrot cake

* Bread, burgers

* Debugging tasting

* Always add the reagents properly, I guess

* Finally got nutriment scaling working

* Finally got all the stuff working and mixing

* PEOPLE CAN NOW TASTE THINGS

* Mouthful only has one L

* Meat n dough

* Eggs and cakes

* Meat dishes taste of meat

* Others...

* Pastry... there is no end to the food

* PIE AND PIZZA

* SANWHICHES, SALAD, SOUP

* THE LAST OF THE FOODS ARE COMPLETE

* Weird indentation issues fixed

* Remove dat debug code

* Hydroponics and sugar is less strong

* Replaces carrot cake reagents

* Code review

* Fixes bad flavours

tgstation-server committed:
Automatic changelog generation for PR #24323 [ci skip]

silicons committed:
[s] Throttles action buttons to one click per decisecond. (#24343)

* Update mob_defines.dm

* Update action_button.dm

* Update action_button.dm

* next click

* Update action_button.dm

* Update action_button.dm

MrsTonedOne committed:
Parallax defaults to disabled on 510 clients. (#24347)

* Update preferences.dm

* Update preferences_savefile.dm

* Update parallax.dm

* Update parallax.dm

client -> C

ChangelingRain committed:
Off-station and carded AIs no longer prevent Judgement scripture from unlocking (#24337)

tgstation-server committed:
Automatic changelog generation for PR #24337 [ci skip]

WJohn committed:
Merge pull request #24348 from dannno/vendy

Sodie pops

tgstation-server committed:
Automatic changelog generation for PR #24348 [ci skip]

lzimann committed:
Fixes blobspore runtimes on Destroy (#24377)

XDTM committed:
Fixes Voice of God (#24328)

AnturK committed:
Merge pull request #24368 from Cyberboss/LiterallyOnFire

Until I figure out why random keys aren't being transferred

Commit:
trendy vendies (#24384)

optimumtact committed:
Merge pull request #24329 from Tofa01/cryosound-master

Disables Sound Variance On Cryo Tube Alert Sound

tgstation-server committed:
Automatic changelog generation for PR #24329 [ci skip]

Cyberboss committed:
Asay and dsay radio keys (#24341)

:cl: Cyberboss
add: Admins can now asay with ":p" and dsay with ":d"
/:cl:

tgstation-server committed:
Automatic changelog generation for PR #24341 [ci skip]

optimumtact committed:
Merge pull request #24342 from ChangelingRain/researchscanfix

You can now have multiple research scanners active at a time

ChangelingRain committed:
Clockcult fixes (#24313)

* Clockcult fixes

* Silicons had identical stun procs for some reason

* oh.

* proper!

optimumtact committed:
Merge pull request #24295 from Nienhaus/sprites

Updates 40mm, 9mm, 10mm, and 45mm sprites

tgstation-server committed:
Automatic changelog generation for PR #24295 [ci skip]

Cyberboss committed:
[READY]Forwarding of New arguments to Initialize (#24298)

* Forwarding of New arguments to Initialize

* Updates that line about New/Initialize arguments

vcordie committed:
Adds the mech tesla cannon (#24252)

* Adds the mech tesla cannon

* makes it not stun

* fixes a dumb

* fixes another dumb

* makes it not blow fucking doors up instantly

* corrects an oversight

* Revert "corrects an oversight"

This reverts commit 73c713a3649deee6945f005f3c14ec926c9559f2.

Commit:
Automatic changelog compile, [ci skip]

Commit:
add flavor fix poster

Commit:
fix red cola

ChangelingRain committed:
Merge pull request #24389 from dannno/HELP_HELP_HELP_HELP

add flavor fix poster

Commit:
i crave release

ChangelingRain committed:
Merge pull request #24392 from dannno/please_just_put_me_down_like_the_dog_i_am

no you're fine

MrsTonedOne committed:
Lagchecks removing from storage, reverts bluespace ore satchel limit (#24310)

tgstation-server committed:
Automatic changelog generation for PR #24310 [ci skip]

Commit:
Automatic changelog compile, [ci skip]

Commit:
try again

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into more_3/4

# Conflicts:
#	code/game/machinery/cloning.dm

Commit:
fixes kommisar konflikts and the final hopefully commit to the purge of non-3/4 boxes

WJohn committed:
Merge pull request #24307 from Militaires/more_3/4

3/4 sprite boxes REDONE

tgstation-server committed:
Automatic changelog generation for PR #24307 [ci skip]

AnturK committed:
Merge pull request #24331 from Cyberboss/SortedClientTimers

[READY]Sorted client time timers

CryoSpacewalk committed:
add science crates from /vg/ (#24349)

tgstation-server committed:
Automatic changelog generation for PR #24349 [ci skip]

KorPhaeron committed:
Fixes mutant eyes (#24338)

* Eyeballs for mutants

* Removes redundant lines

Commit:
with myself (#24411)

optimumtact committed:
Merge pull request #24358 from Cyberboss/DoesAThingForRemie

[READY]Cleans up some code

optimumtact committed:
Merge pull request #24366 from ChangelingRain/cultname

Cult messages will now always include the user's real name

Commit:
[Delta] Adds A Centcomm Ferry Shuttle & [All Maps] Changes Centcom Ferry To Better Design (#24353)

* Adds Centcomm ferry shuttle to Deltastation

* Changes look of CC ferry on deltastation and increases docking hight on Z2 (CC Station) by one to accommodate new changes

* Adds new Centcom  ferry shuttle to all maps.

* Gives centcomm ferry dock a name

* changes chair arrangement

tgstation-server committed:
Automatic changelog generation for PR #24353 [ci skip]

ChangelingRain committed:
Moves His Grace's sprites to weapons.dmi (#24332)

* Moves His Grace's sprites to weapons.dmi

* hup

* here's my compromise

Robustin committed:
Hulks can no longer use Pneumatic Cannons or Flamethrowers (#24383)



Edit: Gave flamethrowers back to drones #ThanksKor
Edit2: This also includes the species check that locks out ash lizards and golems I think

🆑 Robustin
tweak: Hulks can no longer use pneumatic cannons or flamethrowers
/🆑

tgstation-server committed:
Automatic changelog generation for PR #24383 [ci skip]

AnonymousNow committed:
Adds Jamjar Glasses (#24373)

Now's PR experiments #3: Adding an item to a vending machine. Part of a series of tests for me to figure out how the code works a little better.

These are Jamjar Glasses, so called because they're as big and thick as the bottom of jam jars. They're a comical "fashion" item alternative to regular prescription glasses, with all the same functionality. They appear in the Clothesmate vending machine, where they replace one of two existing pairs of prescription glasses.
glasses

🆑
add: Nerd Co. has sent pairs of thicker prescription glasses out to Nanotrasen stations, for your local geek to wear.
/🆑

tgstation-server committed:
Automatic changelog generation for PR #24373 [ci skip]

coiax committed:
Adds plastic, makes plastic flaps constructable and deconstructable (#24257)

* Adds plastic, makes platic flaps constructable

:cl: coiax
add: Cargo can now order plastic sheets to make plastic flaps. No doubt
other uses for plastic will be discovered in the future.
/:cl:

Later maybe we can make cable coils use plastic and metal, rather than
metal and glass. And make plastic tables. And make foam darts use
plastic rather than metal. And make spray bottles out of plastic. And
cheap plastic tables. And cheap plastic chairs.

Why? Because plastic makes sense as a material that you can make stuff
out of, and those DAMNED PLASTIC FLAPS AREN'T CONSTRUCTABLE.

* New plastic sheets sprite (STOLEN FROM BAY)

* Making plastic flaps takes time

* Constructable flaps

* Deconstruction time like walls

tgstation-server committed:
Automatic changelog generation for PR #24257 [ci skip]

MrsTonedOne committed:
fix parallax (#24410)

also fixes some client lag with parallax since the overlay update.

before the overlay update, this would have set the overlay list to the new overlay list (basically doing what the overlay update does for all overlays, but manually). 

Speaking of that, we need a replace_overlays or set_overlays function.

Commit:
Automatic changelog compile, [ci skip]

Jordie0608 committed:
Merge branch 'revert-24324-revert-24277-doesanindexedindexofallindexesindexitself' of https://github.com/tgstation/-tg-station into revert-24324-revert-24277-doesanindexedindexofallindexesindexitself

Jordie0608 commented:
@MrStonedOne I got very close to success but I've been stumped by a single fault in testing
While using `CASE()` to filter test values for the port like so returns as expected:
![capture](https://cloud.githubusercontent.com/assets/4343468/23331688/cd6b1b0e-fbbe-11e6-9bfb-68ad93e93adf.PNG)
Howerver when updating into column `ADD COLUMN server_port_t SMALLINT UNSIGNED NOT NULL`
```
UPDATE `feedback`.`ban2` SET `server_port_t` = CASE
	WHEN NULLIF(CAST(SUBSTRING_INDEX(`serverip`, ':', -1) AS UNSIGNED), '') IS NULL THEN '0'
	WHEN NULLIF(CAST(SUBSTRING_INDEX(`serverip`, ':', -1) AS UNSIGNED), CAST(SUBSTRING_INDEX(`serverip`, '.', 1) AS UNSIGNED)) IS NULL THEN '0'
	ELSE CAST(SUBSTRING_INDEX(`serverip`, ':', -1) AS UNSIGNED)
END
```
I instead get `Error Code: 1292. Truncated incorrect INTEGER value: ':2337'`
I'll try some more when I wake up; As of 1am I've got no clue what's wrong here but I hope it isn't something really elementary.

MrStonedOne commented:
you found `nullif`, but not `ifnull`?

```sql
SELECT serverip, ifnull(INET_ATON(substring_index(serverip, ':', 1)), 0), 
ifnull(CAST(substring_index(serverip, ':', -1) AS UNSIGNED INTEGER), 0) FROM `ss13ban` group by serverip
```
this seems to properly work, i'll try the update after i reset the testdb.

Jordie0608 committed:
edits to text and better update queries
